homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -520,7 +523,9 @@ async def put(self, request, username, entity_number):
             # they'll map to "on". Thus, instead of reporting its actual
             # status, we report what Alexa will want to see, which is the same
             # as the actual requested command.
-            config.cached_states[entity_id] = parsed
+            config.cached_states[entity_id] = [parsed, None]
+        else:
+            config.cached_states[entity_id] = [parsed, time.time()]
```

Comment:
This is going to obscure any failures.  We can call `async_track_state_change_event` and store the callable it returns in the cache as well as an `asyncio.Event`.  Once the state change fires we can set the event.  Then in the `get_entity_state` we can wait for the event for up to the timeout (maybe 2 seconds) and verify the entity state matches the cache (ignoring brightness).  If it matches, return the cache, if it doesn't match return the current state.

Reply:
I don't think that obscuring any failures is what you want if you want to properly mimic how a Hue hub work.
A real hue device never reports a failure from my testing.

Here is a command with a Hue and a zigbee bulb that has been off for over 10 minutes.
The Hue app will show "All lights are on" or "All lights are off" for the room containing that bulb, even with the light showing up as "unreachable".

Sending the following command over the Hue API will show:
```
$ curl -X PUT -d '{"on":true, "bri":254}' http://192.168.10.180/api/XXX/lights/20/state  && echo && curl http://192.168.10.180/api/XX/lights/20
[{"success":{"/lights/20/state/on":true}},{"success":{"/lights/20/state/bri":254}}]
{"state":{"on":true,"bri":254,"alert":"select","mode":"homeautomation","reachable":false},"swupdate":{"state":"notupdatable","lastinstall":"2020-08-02T03:28:21"},"type":"Dimmable light","name":"Dimmable light 1","modelid":"TRADFRI bulb E27 W opal 1000lm","manufacturername":"IKEA of Sweden","productname":"Dimmable light","capabilities":{"certified":false,"control":{},"streaming":{"renderer":false,"proxy":false}},"config":{"archetype":"classicbulb","function":"functional","direction":"omnidirectional"},"uniqueid":"00:0d:6f:ff:fe:fc:f5:e3-01","swversion":"1.2.214"}
```
As you can see, the state and brightness are reported exactly as what the last command was set; however the reachable attribute is set to false. There's no link between the state and if the light was indeed reachable.

The "reachable" attribute is set independently of the cached value as it's set when creating the Hue json.

So I believe the new behaviour set by this changeset is the correct one, and it gives the closest replicate of the original Hue hub behaviour.

Reply:
For comparison, here is what the emulated hue will return under the same circumstances:
```
$ curl -X PUT -d '{"on":true,"bri":254}' 192.168.10.5/api/XXX/lights/56/state && echo && curl http://192.168.10.5/api/XXX/lights/56
[{"success": {"/lights/56/state/on": true}}, {"success": {"/lights/56/state/bri": 254}}]
{"manufacturername": "Home Assistant", "modelid": "HASS123", "name": "Kitchen Light 3", "state": {"bri": 254, "mode": "homeautomation", "on": true, "reachable": false}, "swversion": "123", "type": "Dimmable light", "uniqueid": "00:04:db:70:6d:55:14:d8-9d"}
```

As you can see, it's similar to the real Hue, a set and brightness set to the PUT arguments; but a reachable attribute set to false.

Reply:
Right now alexa is reporting failures when the values don't match.  I don't think we should break that functionality.